asv_bench/benchmarks/omnisci/benchmarks.py

```diff
@@ -29,20 +29,63 @@
     get_benchmark_shapes,
 )
 
+import numpy as np
+
 
 class TimeJoin:
-    param_names = ["shape", "how"]
+    param_names = ["shape", "how", "is_equal_keys"]
     params = [
         get_benchmark_shapes("omnisci.TimeJoin"),
         ["left", "inner"],
+        [True, False],
     ]
 
-    def setup(self, shape, how):
-        self.df1 = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
-        self.df2 = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
+    def setup(self, shape, how, is_equal_keys):
+        self.df1 = generate_dataframe(
+            ASV_USE_IMPL,
+            "int",
+            *shape,
+            RAND_LOW,
+            RAND_HIGH,
+            cache_prefix="left_join_frame",
+        )
+        self.df2 = generate_dataframe(
+            ASV_USE_IMPL,
+            "int",
+            *shape,
+            RAND_LOW,
+            RAND_HIGH,
+            cache_prefix="right_join_frame",
+        )
+        if is_equal_keys:
+            # OmniSci fails to join on a 'rowid' (backend uses internal row indices
+            # as keys to join if an actual index isn't specified), so explicitly
+            # specifying an index to join on it.
+            # https://github.com/modin-project/modin/issues/3442
+            common_index = np.arange(1, len(self.df1) + 1)
```

Comment:
Rephrased in-code comment:
```python
            # When the frames have default indices to join on: RangeIndex(frame_length),
            # OmniSci backend performs join on the internal meta-column called 'rowid'.
            # There is a bug in the engine that makes such joins fail. To avoid joining
            # on the meta-column we explicitly specify a non-default index to join on.
            # https://github.com/modin-project/modin/issues/3442
```